### PR TITLE
Updating the Java Frontend project template to work with Gradle 7.0

### PR DESCRIPTION
**Pull request by JaroslavTulach**
Pull request #2999 had to fix the version of Gradle for HTML/Java project template 6.9 due to incompatibilities in retrolamda plugin. Updating the project template to work on Gradle 7.x by [removing the need of retrolamda](http://wiki.apidesign.org/wiki/RetroLambda).

### CODE REVIEW

File: java/gradle.htmlui/src/org/netbeans/modules/gradle/htmlui/resources/root_build.gradle.fmk [hunk 1]
```diff
@@ -18,15 +18,14 @@
     under the License.
 
 -->
-<#noparse>
-plugins {
-    id "me.tatarka.retrolambda" version "3.7.1"
```
Comment (lkishalmi) [inline java/gradle.htmlui/src/org/netbeans/modules/gradle/htmlui/resources/root_build.gradle.fmk:23]: Just out of curiosity, is retrolambda no longer required?

**Review by lkishalmi — APPROVED** [review]
Looks fine.

File: java/gradle.htmlui/src/org/netbeans/modules/gradle/htmlui/resources/root_build.gradle.fmk (hunk 1 above)
Comment (JaroslavTulach) [reply java/gradle.htmlui/src/org/netbeans/modules/gradle/htmlui/resources/root_build.gradle.fmk:23]: Right, [retrolambda is no longer required](http://wiki.apidesign.org/wiki/Bck2Brwsr#Version_0.50).

**Comment by JaroslavTulach** [comment]
Opps, 12.6!? This fix should rather go into 12.5! I guess I prepare one more PR.

**Comment by neilcsmith-net** [comment]
A PR could make syncing delivery to master difficult for @ebarboni If this needs to be in 12.5 then some git-fu may be required?

**Comment by matthiasblaesing** [comment]
I would expect this to go cleanly, if the commit is merged (cleanly) with `git cherry-pick`. As long as there are no conflicting changes in both branches this should be save. If `delivery` is merged shortly (IMHO delivery -> master merges can be done pretty much immediately) this should be handable.

**Comment by neilcsmith-net** [comment]
Yes, that sounds about right. Needs care.

Delivery -> master sync PRs are usually done immediately after each beta and release - ie. sync delivery to release, check build, sync delivery to master.

**Comment by JaroslavTulach** [comment]
> I would expect this to go cleanly, if the commit is merged (cleanly) with `git cherry-pick`. 

Backported the commit as 0728145 and added few commits on top of it. Hopefully the merge of `delivery` to `master` will be smooth.

**Pull request merged**